# Data Analysis: Video Games
## Chris Turner - Aug 30, 2018




In [1]:
import csv
import pandas as pd
import numpy as np
import datetime as dt


In [2]:
df = pd.read_csv('./assets/vgsales.csv')

In [3]:
df = df[df.Global_Sales > .1]

## 1. Which company is the most common video game publisher?


In [4]:
counted_pub = df.Publisher.groupby(df.Publisher).count()


In [5]:
pd.DataFrame(counted_pub.nlargest(5))

Publisher
Publisher                 
Electronic Arts       1161
Activision             780
Ubisoft                654
Nintendo               631
THQ                    585

### Electronic Arts! 

## 2. What’s the most common platform?

In [6]:
counted_plat = df.Platform.groupby(df.Platform).count()

In [7]:
pd.DataFrame(counted_plat.nlargest(5))

Platform
Platform          
PS2           1520
DS            1099
PS3           1000
X360           971
Wii            913

### PS2!

## 3. What about the most common genre?


In [8]:
counted_genre = df.Genre.groupby(df.Genre).count()

In [9]:
pd.DataFrame(counted_genre.nlargest(5))

Genre
Genre              
Action         2201
Sports         1695
Misc           1071
Role-Playing    951
Shooter         895

### Action

## 4. What are the top 20 highest grossing games?


In [10]:
pd.DataFrame(df.head(20))

Rank                                          Name Platform    Year  \
0      1                                    Wii Sports      Wii  2006.0   
1      2                             Super Mario Bros.      NES  1985.0   
2      3                                Mario Kart Wii      Wii  2008.0   
3      4                             Wii Sports Resort      Wii  2009.0   
4      5                      Pokemon Red/Pokemon Blue       GB  1996.0   
5      6                                        Tetris       GB  1989.0   
6      7                         New Super Mario Bros.       DS  2006.0   
7      8                                      Wii Play      Wii  2006.0   
8      9                     New Super Mario Bros. Wii      Wii  2009.0   
9     10                                     Duck Hunt      NES  1984.0   
10    11                                    Nintendogs       DS  2005.0   
11    12                                 Mario Kart DS       DS  2005.0   
12    13                   Pokemon Gold/Pokemon Silver       GB  1999.0   
13    14                                       Wii Fit      Wii  2007.0   
14    15                                  Wii Fit Plus      Wii  2009.0   
15    16                            Kinect Adventures!     X360  2010.0   
16    17                            Grand Theft Auto V      PS3  2013.0   
17    18                 Grand Theft Auto: San Andreas      PS2  2004.0   
18    19                             Super Mario World     SNES  1990.0   
19    20  Brain Age: Train Your Brain in Minutes a Day       DS  2005.0   

           Genre               Publisher  NA_Sales  EU_Sales  JP_Sales  \
0         Sports                Nintendo     41.49     29.02      3.77   
1       Platform                Nintendo     29.08      3.58      6.81   
2         Racing                Nintendo     15.85     12.88      3.79   
3         Sports                Nintendo     15.75     11.01      3.28   
4   Role-Playing                Nintendo     11.27      8.89     10.22   
5         Puzzle                Nintendo     23.20      2.26      4.22   
6       Platform                Nintendo     11.38      9.23      6.50   
7           Misc                Nintendo     14.03      9.20      2.93   
8       Platform                Nintendo     14.59      7.06      4.70   
9        Shooter                Nintendo     26.93      0.63      0.28   
10    Simulation                Nintendo      9.07     11.00      1.93   
11        Racing                Nintendo      9.81      7.57      4.13   
12  Role-Playing                Nintendo      9.00      6.18      7.20   
13        Sports                Nintendo      8.94      8.03      3.60   
14        Sports                Nintendo      9.09      8.59      2.53   
15          Misc  Microsoft Game Studios     14.97      4.94      0.24   
16        Action    Take-Two Interactive      7.01      9.27      0.97   
17        Action    Take-Two Interactive      9.43      0.40      0.41   
18      Platform                Nintendo     12.78      3.75      3.54   
19          Misc                Nintendo      4.75      9.26      4.16   

    Other_Sales  Global_Sales  
0          8.46         82.74  
1          0.77         40.24  
2          3.31         35.82  
3          2.96         33.00  
4          1.00         31.37  
5          0.58         30.26  
6          2.90         30.01  
7          2.85         29.02  
8          2.26         28.62  
9          0.47         28.31  
10         2.75         24.76  
11         1.92         23.42  
12         0.71         23.10  
13         2.15         22.72  
14         1.79         22.00  
15         1.67         21.82  
16         4.14         21.40  
17        10.57         20.81  
18         0.55         20.61  
19         2.05         20.22

## 5. For North American video game sales, what’s the median?
### Provide a secondary output showing ‘about’ ten games surrounding the median sales output


In [11]:
american_median = df.NA_Sales.median()


### 0.18

In [12]:
df.iloc[(df['NA_Sales']-american_median).abs().argsort()[:11]]

Rank                                               Name Platform  \
7693  7695             Puzzle Challenge: Crosswords and More!      PSP   
6587  6589                             ATV: Quad Power Racing      GBA   
5703  5705                         WarCraft II: The Dark Saga       PS   
4521  4523                                    Mega Man Zero 2      GBA   
4513  4515                                          Minecraft     WiiU   
6580  6582                       Bionicle: Matoran Adventures      GBA   
6835  6837                                        Kill.Switch       XB   
4455  4457                                       Naughty Bear      PS3   
5762  5764                                Disney Infinity 3.0     XOne   
6562  6564  Dora the Explorer: Game Boy Advance Video Volu...      GBA   
5765  5767                                   MySims SkyHeroes      Wii   

        Year     Genre                   Publisher  NA_Sales  EU_Sales  \
7693  2006.0    Puzzle         Crave Entertainment      0.18      0.00   
6587  2002.0    Racing                Liquid Games      0.18      0.07   
5703  1997.0  Strategy             Electronic Arts      0.18      0.12   
4521  2003.0  Platform                      Capcom      0.18      0.07   
4513  2016.0      Misc      Microsoft Game Studios      0.18      0.09   
6580  2002.0  Platform             Electronic Arts      0.18      0.07   
6835  2003.0   Shooter          Namco Bandai Games      0.18      0.05   
4455  2010.0    Action                   505 Games      0.18      0.18   
5762  2015.0    Action  Disney Interactive Studios      0.18      0.10   
6562  2004.0      Misc                         NaN      0.18      0.07   
5765  2010.0    Action             Electronic Arts      0.18      0.10   

      JP_Sales  Other_Sales  Global_Sales  
7693      0.00         0.01          0.20  
6587      0.00         0.00          0.26  
5703      0.00         0.02          0.31  
4521      0.17         0.01          0.43  
4513      0.14         0.03          0.43  
6580      0.00         0.00          0.26  
6835      0.00         0.01          0.24  
4455      0.00         0.07          0.44  
5762      0.00         0.03          0.31  
6562      0.00         0.00          0.26  
5765      0.00         0.03          0.31

## 6. For the top-selling game of all time, how many standard deviations above/below the mean are its sales for North America?

In [13]:
standard_dev_na = df.NA_Sales.std()
na_mean = df.NA_Sales.mean()
largest = df.NA_Sales.nlargest(1)[0]

In [14]:
dev_over = (largest - na_mean ) / standard_dev_na

### 40.95953868369813

## 7. The Nintendo Wii seems to have outdone itself with games. How does its average number of sales compare with all of the other platforms?


In [15]:
df.Global_Sales.groupby(df.Platform).mean()['Wii'] / [df.Global_Sales.groupby(df.Platform).mean().sum() - df.Global_Sales.groupby(df.Platform).mean()['Wii']][0]

0.04275663968092074

## Come up with 3 more questions that can be answered with this data set.

## 1. What was the worst selling platform over all? 

In [17]:
counted_plat = df.Platform.groupby(df.Platform).count()
pd.DataFrame(counted_plat.nsmallest(5))

Platform
Platform          
TG16             1
SCD              2
WS               5
NG               6
GEN             17

### TurboGrafx-16 is the worst with only 1 game in the list

## 2. What was the peek year for global sales?

In [21]:
games_per_year_per_platform = df.groupby('Year')['Global_Sales'].sum()

In [19]:
games_per_year_per_platform

Year
1980.0      9
1981.0     46
1982.0     36
1983.0     17
1984.0     14
1985.0     12
1986.0     21
1987.0     14
1988.0     14
1989.0     16
1990.0     16
1991.0     37
1992.0     36
1993.0     46
1994.0     86
1995.0    131
1996.0    193
1997.0    231
1998.0    267
1999.0    274
2000.0    271
2001.0    359
2002.0    540
2003.0    521
2004.0    520
2005.0    614
2006.0    520
2007.0    710
2008.0    876
2009.0    866
2010.0    751
2011.0    687
2012.0    394
2013.0    346
2014.0    358
2015.0    284
2016.0    121
2020.0      1
Name: Platform, dtype: int64

### 2008 with 876

## 3. What was the slowest year for games? 

### 1980 with 9